azurelinuxagent/common/protocol/wire.py

```diff
@@ -153,10 +153,9 @@ def get_ext_handlers(self):
         # In wire protocol, incarnation is equivalent to ETag
         return ext_conf.ext_handlers, goal_state.incarnation
 
-    def get_ext_handler_pkgs(self, ext_handler):
+    def get_ext_handler_pkgs(self, ext_handler, etag):
         logger.verbose("Get extension handler package")
-        goal_state = self.client.get_goal_state()
-        man = self.client.get_ext_manifest(ext_handler, goal_state)
+        man = self.client.get_ext_manifest(ext_handler, etag)
```

Comment:
Why remove the call to `get_goal_state()` here? It is simply loading from the disk cache.

Reply:
The only part of the goal state which is relevant is the incarnation number (the etag). The ExtHandlersHandler uses this number to keep track of whether the extensions are current or not anyways. It is more consistent to pass this in for each ExtHandler rather than load the entire goal state IMO.